pkg/helpers/cryptoutils.go

```diff
@@ -76,9 +80,29 @@ func GenerateSelfSignedCA(keyType x509.PublicKeyAlgorithm, expirationTime time.D
 
 // defined to generate certificates with RSA and ECDSA keys
 func GenerateCertificateRequest(subject models.Subject, key any) (*x509.CertificateRequest, error) {
+	template := x509.CertificateRequest{
+		Subject: SubjectToPkixName(subject),
+	}
 
+	csrBytes, err := x509.CreateCertificateRequest(rand.Reader, &template, key)
+	if err != nil {
+		return nil, err
+	}
+
+	csr, err := x509.ParseCertificateRequest(csrBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return csr, err
+}
+
+// defined to generate certificates with RSA and ECDSA keys
+func GenerateCertificateRequestWithExtensions(subject models.Subject, extensions []pkix.Extension, key any) (*x509.CertificateRequest, error) {
 	template := x509.CertificateRequest{
 		Subject: SubjectToPkixName(subject),
+		// ExtraExtensions: extensions,
+		IPAddresses: []net.IP{net.IPv4(192, 168, 100, 1)},
```

Comment:
Hi @haritzsaiz, Is this hardcoded IP really needed?

Reply:
Yes, It is intended to be added as a SAN in the certificate.